### PR TITLE
Rework plugin brokering to reuse PVC strategies code

### LINKED ISSUE #10879: Add an ability to use different PVC strategies in Walking skeleton instead of hardcoding behavior of common strategy
### Description
In the [initial implementation](https://github.com/eclipse/che/compare/master...garagatyi:integrateBroker) of walking skeleton I hardcoded behavior identical to common PVC strategy. 
We need to rework it to avoid bugs if the implementation of this strategy changes. We also need to support unique strategy. 

### Reproduction Steps

**OS and version:**    

**Diagnostics:**

### PR DESCRIPTION
**Pull request by garagatyi**
### What does this PR do?
Rework plugin broker code to reuse PVC strategies code.
Moreover, it applies other OS/K8s provisioners with proxy, name uniqueness, private docker image registry, pod termination support. 

### What issues does this PR fix or reference?
Fixes #10879

### CODE REVIEW

**Comment by garagatyi** [comment]
ci-test

File: infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/wsplugins/brokerphases/KBrokerEnvironmentConfig.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+/**
+ * Extends {@link BrokerEnvironmentConfig} to be used in the kubernetes infrastructure.
+ *
+ * <p>This API is in <b>Beta</b> and is subject to changes or removal.
+ *
+ * @author Oleksandr Garagatyi
+ */
+@Beta
+public class KBrokerEnvironmentConfig extends BrokerEnvironmentConfig<KubernetesEnvironment> {
```
Comment (skabashnyuk) [inline infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/wsplugins/brokerphases/KBrokerEnvironmentConfig.java]: what about names K8sBrokerEnvironmentConfig or KubernetesBrokerEnvironmentConfig?

File: infrastructures/openshift/src/main/java/org/eclipse/che/workspace/infrastructure/openshift/wsplugins/brokerphases/OBrokerEnvironmentConfig.java [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+/**
+ * Extends {@link KBrokerEnvironmentConfig} to be used in the openshift infrastructure.
+ *
+ * <p>This API is in <b>Beta</b> and is subject to changes or removal.
+ *
+ * @author Oleksandr Garagatyi
+ */
+@Beta
+public class OBrokerEnvironmentConfig extends BrokerEnvironmentConfig<OpenShiftEnvironment> {
+
```
Comment (skabashnyuk) [inline infrastructures/openshift/src/main/java/org/eclipse/che/workspace/infrastructure/openshift/wsplugins/brokerphases/OBrokerEnvironmentConfig.java]: OpenShiftBrokerEnvironmentConfig?

**Review by sleshchenko — APPROVED** [review]
Other LGTM

**Comment by riuvshin** [comment]
ci-test build report:
[Build details](https://ci.codenvycorp.com/job/che-pullrequests-test-ocp/762/)
[Test report](https://ci.codenvycorp.com/job/che-pullrequests-test-ocp/762//Selenium_20tests_20report/)
[selenium tests report data](http://tests.codenvycorp.com/selenium_test_report_data/11119)
docker image: eclipseche/che-server:11119
https://github.com/orgs/eclipse/teams/eclipse-che-qa please check this report.

**Review by skabashnyuk — APPROVED** [review]

**Comment by artaleks9** [comment]
Selenium tests execution on Eclipse Che Multiuser on OCP (https://ci.codenvycorp.com/job/che-pullrequests-test-ocp/762/) doesn't show any regression against this Pull Request.

**Pull request merged**